### PR TITLE
Make modals client-side

### LINKED ISSUE #19: Badge field in modal needs autofocus
Badge field needs autofocus and auto submit when filled. Field must be password field so the browser doesn't remember the history of entered badges.

### PR DESCRIPTION
**Pull request by starcraft66**
Refactors modals to be entirely client-side using Alpine.js and the modern `<dialog>` tag instead of being driven by Phoenix LiveView and deprecated DaisyUI functionality. This makes them more responsive and fixes things like auto-selecting the badge serial number field.

Upgraded DaisyUI to 4.6.0 (latest) while testing to see if it fixed a bug but kept it in the PR cause why not.

Updated flake inputs to their latest revisions because there was an stdenv-darwin-related problem preventing me from testing the nix build.

Closes #19

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -101,6 +101,7 @@
       systems = [
         # systems for which you want to build the `perSystem` attributes
         "x86_64-linux"
+        "x86_64-darwin"
```
Comment (starcraft66) [inline flake.nix:104]: This is left-over from when I was developing on my old intel mac but this should still be a supported development platform

File: lib/lanpartyseating_web/components/cancellation_modal.ex [hunk 2]
```diff
@@ -10,110 +10,142 @@ defmodule CancellationModalComponent do
...
+
+                  <%= if !is_nil(@error) do %>
+                    <p class="text-error"><%= @error %></p>
+                  <% end %>
+                  <br/><br/>
+
+                  <div class="modal-action">
+                    <button
+                      class="btn btn-success"
+                      x-on:click={"$refs.station_modal_#{@station.station_number}.close()"}
```
Comment (starcraft66) [inline lib/lanpartyseating_web/components/cancellation_modal.ex:39]: For some reason I can't figure out, all of the modals on this page require a hook to close the modal on the submit button while other pages like the selfsign page simply close the modal just fine when submitting the form. This is an intermediary solution that works without any apparent issues on my end. This comment applies to the rest of the modals in this file too.

File: mix.exs [hunk 3]
```diff
@@ -37,7 +37,7 @@ defmodule Lanpartyseating.Mixfile do
       {:ecto_sql, "~> 3.10.1"},
       {:postgrex, ">= 0.0.0"},
       {:phoenix_html, "~> 3.3.1"},
-      {:phoenix_live_view, "~> 0.19.3"},
+      {:phoenix_live_view, "~> 0.20.0"},
```
Comment (starcraft66) [inline mix.exs:40]: Required upgrade to handle `<form method="dialog">` submissions without crashing the live view connection, see [this commit](https://github.com/phoenixframework/phoenix_live_view/commit/3f33e6dec7c047b19d4af472ebc4019f4c6988dd)

File: priv/repo/seeds.exs [hunk 4]
```diff
@@ -104,3 +104,9 @@ Lanpartyseating.Repo.insert!(%Lanpartyseating.Setting{
   horizontal_trailing: 1,
   vertical_trailing: 0
 })
+
+Lanpartyseating.Repo.insert!(%Lanpartyseating.Badge{
```
Comment (starcraft66) [inline priv/repo/seeds.exs:108]: Convenience seeder to allow one to simply input `1` as the badge number in the selfsign page to easily test station registrations in dev

File: assets/js/app.js [hunk 5]
```diff
@@ -1,8 +1,23 @@
...
+window.Alpine = Alpine
+Alpine.plugin(focus)
+Alpine.start()
 
 let csrfToken = document.querySelector("meta[name='csrf-token']").getAttribute("content")
-let liveSocket = new LiveSocket("/live", Socket, {params: {_csrf_token: csrfToken}})
+let liveSocket = new LiveSocket("/live", Socket, {
+  params: { _csrf_token: csrfToken },
+  dom: {
+    onBeforeElUpdated(from, to) {
```
Comment (starcraft66) [inline assets/js/app.js:14]: Little trick to keep Alpine.js up-to-date with the DOM patches Phoenix LiveView is performing, taken from [here](https://fullstackphoenix.com/tutorials/combine-phoenix-liveview-with-alpine-js)

**Review by luxor37 — APPROVED** [review]
Looks fine to me :)

**Pull request merged**